fourier/fourier_test.go

```diff
@@ -214,6 +214,153 @@ func TestCmplxFFT(t *testing.T) {
 	})
 }
 
+func TestDCT(t *testing.T) {
+	const tol = 1e-10
+	t.Run("NewDCT", func(t *testing.T) {
+		for n := 2; n <= 200; n++ {
+			dct := NewDCT(n)
+
+			want := make([]float64, n)
+			for i := range want {
+				want[i] = rand.Float64()
+			}
+
+			coeff := dct.Transform(nil, want)
+			got := dct.Transform(nil, coeff)
+			floats.Scale(1/float64(2*(n-1)), got)
+
+			if !floats.EqualApprox(got, want, tol) {
+				t.Errorf("unexpected result for transform(transform(x)) for length %d", n)
+			}
+		}
+	})
+	t.Run("Reset DCT", func(t *testing.T) {
+		dct := NewDCT(1000)
+		for n := 2; n <= 2000; n++ {
+			dct.Reset(n)
```

Comment:
Maybe add one test at the end where you reset to a smaller size?

Reply:
Functionally this is what happens. In each case the dct is initially created with 1000, and then increated from the smallest value it can take.